systemd/usr/bin/github-act-runner.sh

```diff
@@ -0,0 +1,451 @@
+    $systemctl_cmd --quiet disable $restarter_service_file
+
+    $systemctl_cmd --quiet stop $runner_service_file
+    $systemctl_cmd --quiet disable $runner_service_file
+}
+
+function assert_runner_exists {
+    local id=$1
+
+    local runner_dir=${runners_dir}${id}
+
+    # check that runner exists
+    [ -d "${runner_dir}" ] || error "runner 'id = ${id}' does not exist"
+}
+
+function handle_new_command {
+    # define required options to empty values
+    declare -A local opts=( \
+        [owner]= \
+        [name]= \
+        [token]= \
+    )
+
+    while [[ $# > 0 ]] ; do
+        case $1 in
+            --help)
+                echo "usage:"
+                echo "	$(basename $0) <...> add <options>"
+                echo ""
+                echo "options:"
+                echo "  --help    show this help text and do nothing."
+                echo "  --owner   github repo (e.g. 'my_user/my_repo') or organization (e.g. 'my_org')."
+                echo "  --name    new runner name."
+                echo "  --labels  comma separated list of runner labels, e.g. 'label1,label1,label3'."
+                echo "  --token   github runner registration token."
+                exit 0
+                ;;
+            --owner)
+                shift
+                opts[owner]=$1
+                ;;
+            --name)
+                shift
+                opts[name]=$1
+                ;;
+            --token)
+                shift
+                opts[token]=$1
+                ;;
+            --labels)
+                shift
+                opts[labels]=$1
+                ;;
+            *)
+                error "unknown option: $1"
+                ;;
+        esac
+        [[ $# > 0 ]] && shift;
+    done
+
+    for opt in ${!opts[@]}; do
+        [ ! -z "${opts[$opt]}" ] || error "missing option: --$opt"
+    done
+
+    if ! $is_root; then
+        # running as non-root user
+        # check that the user is in 'docker' group"
+        if [ -z "$(groups | grep docker)" ]; then
+            error "user '$(id --user --name)' is not in 'docker' group, install docker and add the user to the group"
+        fi
+    fi
+
+    mkdir --parents $runners_dir
+
+    local owner_dot=(${opts[owner]//\//.})
+
+    local runner_id=${opts[name]}.${owner_dot}
+
+    local runner_dir="${runners_dir}${runner_id}"
+    if [ -d "$runner_dir" ]; then
+        error "unable to create new runner: runner 'id = ${runner_id}' already exists"
+    fi
+
+    echo "new runner config will be placed to $runner_dir"
+
+    local labels=
+    if [ ! -z "${opts[labels]}" ]; then
+        # NOTE: that runner_bin has '--label' key, not '--labels', TODO: fix that in runner_bin?
+        labels="--label ${opts[labels]}"
+    fi
+
+    mkdir --parents $runner_dir
+
+    # remove the new runner dir in case of ERROR
+    add_to_err_trap "rm --recursive --force $runner_dir"
+
+    local url=https://github.com/${opts[owner]}
```

Comment:
What if you have a non github.com instance like GHES?

Reply:
I didn't think about that the domain can be different. I don't know anything about github enterprise actually...
But is it always in the form https://some_domain/user/repo?
The thing is that runners are configured per repo or per organization, so there can be runners with same name but in different repos, so we need to add repo name or org name to the final runner `id`. If you know more details about possible corner cases, feel free to propose solution, as I might lack info/knowledge about GHES specifics.

Reply:
> But is it always in the form https://some_domain/user/repo?

Yes the same form but different domain, _it is technically possible that a dev server uses the insecure http protocol_

_This runner wouldn't be possible without my own github actions server https://github.com/ChristopherHX/runner.server, github.com gave me some weird errors during early testing._

Reply:
then, perhaps another CLI option could be added, like `--doman`, so it would be something like:
```console
github-act-runner new --doman https://some_doman.com --owner user/repo --name my_runner --token ...
```
so that there would be no need to parse the whole URL to extract `owner/repo` from it... On the other hand, if there are different domains possible, then domain should also be part of the runner `id`, then runner `id` becomes too long...